src/lib.rs

```diff
@@ -189,6 +191,58 @@ impl Tokenizer {
         })
     }
 
+    /// Tokenize a batch of multiple input strings.
+    ///
+    /// Each given input string is tokenized and the numeric representation
+    /// written to a row in the resulting two-dimensional matrix of shape
+    /// `(texts.len(), context_length)`, with the special `<start_of_text>`
+    /// token prepended, and `<end_of_text>` appended to each text.
+    ///
+    /// The individual input strings are lowercased before being tokenized, but
+    /// otherwise no pre-processing is performed.
+    ///
+    /// `context_length` is the maximum number of tokens per each text and
+    /// should be `77` for all current CLIP models. If tokenizing an input text
+    /// results in too many tokens, the token sequence will be truncated to fit
+    /// within the resulting row of length `context_length`, always including
+    /// the `<start_of_text>` and `<end_of_text>` marker tokens.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # use ndarray::array;
+    /// # use instant_clip_tokenizer::{Token, Tokenizer};
+    /// let tokenizer = Tokenizer::new();
+    /// let encoded = tokenizer.tokenize_batch(["Hi", "How are you?"], 5);
+    /// assert_eq!(encoded, array![
+    ///     [49406, 1883, 49407, 0, 0],
```

Comment:
This makes it seem like the notion of trailing zeros is somehow specialized. Is that the case? How/where is that case handled?

Reply:
> Now, in case the tokenization of an input string results in less than `context_length` tokens the remaining matrix elements need to have some value, and `0` is what the Python code uses as a "padding".

I was talking about this input value stuff. IIRC there is no other case in the Rust library where this padding concept is used, so we should make sure it is adequately documented here. We're building a Rust library here so it's not sufficient to implicitly assume choices from another Python library in the public interface of this library.